libbpf-tools/biosnoop.c

```diff
@@ -162,13 +179,24 @@ void handle_event(void *ctx, int cpu, void *data, __u32 data_sz)
 	const struct partition *partition;
 	const struct event *e = data;
 	char rwbs[RWBS_LEN];
-
-	if (!start_ts)
+	struct timespec ct;
+	struct tm *tm;
+	char ts[32];
+
+	if (env.timestamp) {
+		/* Since `bpf_ktime_get_boot_ns` requires at least 5.8 kernel,
+		 * so get time from usespace instead */
+		clock_gettime(CLOCK_REALTIME, &ct);
+		tm = localtime(&ct.tv_sec);
+		strftime(ts, sizeof(ts), "%H:%M:%S", tm);
+		printf("%-8s.%03ld ", ts, ct.tv_nsec / 1000000);
+	} else if (!start_ts) {
 		start_ts = e->ts;
+		printf("%-11.6f ",(e->ts - start_ts) / 1000000000.0);
+	}
```

Comment:
without timestamp, the output is not correct:
```
$ sudo ./biosnoop
TIME(s)     COMM           PID     DISK    T    SECTOR     BYTES   LAT(ms)
0.000000    kworker        2493382 vda     W    242087024  516096    0.998
kworker        2493382 vda     W    242088032  516096    1.011
kworker        2493382 vda     W    242089040  16384     1.192
kworker        2493382 vda     W    242089072  516096    1.378
kworker        2493382 vda     W    242090080  516096    1.373
...
```

Reply:
Fixed, thanks!